src/Map/assets/src/google_maps_controller.ts

```diff
@@ -0,0 +1,204 @@
+    private infoWindows: Array<google.maps.InfoWindow> = [];
+
+    initialize() {
+        const providerConfig = window.__symfony_ux_maps.providers?.google_maps;
+        if (!providerConfig) {
+            throw new Error(
+                'Google Maps provider configuration is missing, did you forget to call `{{ ux_map_script_tags() }}`?'
+            );
+        }
+
+        const loaderOptions: LoaderOptions = {
+            apiKey: providerConfig.key,
+        };
+
+        this.dispatchEvent('init', {
+            loaderOptions,
+        });
+
+        this.loader = new Loader(loaderOptions);
+    }
+
+    async connect() {
+        const { Map: GoogleMap, InfoWindow } = await this.loader.importLibrary('maps');
+
+        const mapOptions: google.maps.MapOptions = {
+            gestureHandling: this.viewValue.gestureHandling,
+            backgroundColor: this.viewValue.backgroundColor,
+            disableDoubleClickZoom: this.viewValue.disableDoubleClickZoom,
+            zoomControl: this.viewValue.zoomControl,
+            zoomControlOptions: this.viewValue.zoomControlOptions,
+            mapTypeControl: this.viewValue.mapTypeControl,
+            mapTypeControlOptions: this.viewValue.mapTypeControlOptions,
+            streetViewControl: this.viewValue.streetViewControl,
+            streetViewControlOptions: this.viewValue.streetViewControlOptions,
+            fullscreenControl: this.viewValue.fullscreenControl,
+            fullscreenControlOptions: this.viewValue.fullscreenControlOptions,
+        };
+
+        if (this.viewValue.mapId) {
+            mapOptions.mapId = this.viewValue.mapId;
+        }
+
+        if (this.viewValue.center) {
+            mapOptions.center = this.viewValue.center;
+        }
+
+        if (this.viewValue.zoom) {
+            mapOptions.zoom = this.viewValue.zoom;
+        }
+
+        this.dispatchEvent('pre-connect', {
+            mapOptions,
+        });
+
+        this.map = new GoogleMap(this.element, mapOptions);
+
+        if (this.viewValue.markers) {
+            const { AdvancedMarkerElement } = await this.loader.importLibrary('marker');
+
+            this.viewValue.markers.forEach((markerConfiguration) => {
+                const marker = new AdvancedMarkerElement({
+                    position: markerConfiguration.position,
+                    title: markerConfiguration.title,
+                    map: this.map,
+                });
+
+                this.markers.set(markerConfiguration._id, marker);
+            });
+
+            if (this.viewValue.fitBoundsToMarkers) {
+                const bounds = new google.maps.LatLngBounds();
+                this.markers.forEach((marker) => {
+                    if (!marker.position) {
+                        return;
+                    }
+
+                    bounds.extend(marker.position);
+                });
+                this.map.fitBounds(bounds);
+            }
```

Comment:
Fit map's bounds to markers is a feature that a lot of people needs (including myself):
<img width="863" alt="image" src="https://github.com/symfony/ux/assets/2103975/2518af3a-4b01-4caa-80d0-5b8f1ead1060">

Having it built-in is a nice to have IMO